combat-trainer.lic

```diff
@@ -2063,7 +2063,11 @@ class AbilityProcess
       .map { |name| "Khri #{name}" }
       .each do |name|
         next if game_state.danger && DRCA.kneel_for_khri?(@kneel_khri, name)
+        timer = game_state.cooldown_timers[name]
+        next unless !timer || (Time.now - timer).to_i > 30
+
```

Comment:
What happened that the timers were needed? I would think that `spellmonitor` keeping track of your abilities and `DRCA.activate_khri?` using `DRSpells.active_spells` would mean it'd know if the abilities were up or not.

Reply:
Or, is this more of a "don't check to rebuff more of than every 30 seconds"?

Reply:
I think this is added primarily to keep it from re-checking too often for things that succeeded (especially for people who run 'delay <whatever>' in their khri for the slow start + better effect versions).

It might be interesting to try and figure out a 'pending spells' mechanic somewhere in the system so that activate khri w/ delay could say "consider this running for 30 seconds until it really is running", but for now, this timer fixes it for combat, and most people aren't spam running buff so it doesn't matter there.